apps/web/app/api/generate/route.ts

```diff
@@ -53,26 +55,94 @@ export async function POST(req: Request): Promise<Response> {
     }
   }
 
-  let { prompt } = await req.json();
+  let { prompt, option } = await req.json();
 
-  const response = await openai.chat.completions.create({
-    model: process.env.NODE_ENV == "development" ? "llama2" : "gpt-3.5-turbo",
-    stream: true,
-    messages: [
+  const messages = match(option)
+    .with("continue", () => [
       {
         role: "system",
         content:
           "You are an AI writing assistant that continues existing text based on context from prior text. " +
           "Give more weight/priority to the later characters than the beginning ones. " +
-          "Limit your response to no more than 200 characters, but make sure to construct complete sentences.",
-        // we're disabling markdown for now until we can figure out a way to stream markdown text with proper formatting: https://github.com/steven-tey/novel/discussions/7
-        // "Use Markdown formatting when appropriate.",
+          "Limit your response to no more than 200 characters, but make sure to construct complete sentences." +
+          "Use Markdown formatting when appropriate.",
+      },
+      {
+        role: "user",
+        content: prompt,
+      },
+    ])
+    .with("improve", () => [
+      {
+        role: "system",
+        content:
+          "You are an AI writing assistant that improves existing text. " +
+          "Limit your response to no more than 200 characters, but make sure to construct complete sentences." +
+          "Use Markdown formatting when appropriate.",
+      },
+      {
+        role: "user",
+        content: prompt,
+      },
+    ])
+    .with("shorter", () => [
+      {
+        role: "system",
+        content:
+          "You are an AI writing assistant that shortens existing text. " +
+          "Use Markdown formatting when appropriate.",
+      },
+      {
+        role: "user",
+        content: prompt,
+      },
+    ])
+    .with("longer", () => [
+      {
+        role: "system",
+        content:
+          "You are an AI writing assistant that lengthens existing text. " +
+          "Use Markdown formatting when appropriate.",
+      },
+      {
+        role: "user",
+        content: prompt,
```

Comment:
![image](https://github.com/steven-tey/novel/assets/45734348/cddd2ca0-acdd-405f-a915-de89a89896af)
![image](https://github.com/steven-tey/novel/assets/45734348/8599c6a4-0e29-4c94-82ab-ba9d277f66ed)

Do you think we should add further information for content?, I just added simple expand sentence like this
```content: `The existing text is: ${prompt}`,```
The screenshot show improve abit

Reply:
hey, you are right the prompt content is not good enought, I'll add an issue for this after this PR is merged

Reply:
thank @andrewdoro